src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,302 @@
+package duke;
+
+import duke.exception.AtEmptyException;
+import duke.exception.ByEmptyException;
+import duke.exception.NumberOutOfBoundsException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Duke {
+    private static final String HORIZONTAL_LINE = "____________________________________________________________";
+    private static final String INDENT = "    ";
+    private static final int START_OF_STRING = 0;
+    public static final String SAD_FACE = "\uD83D\uDE41";
+
+    private static String inputCommand;
+    private static String todoTask;
+    private static String deadlineDescription;
+    private static String by;
+    private static String eventDescription;
+    private static String at;
+    private static int inputNum;
+    private static String number;
+
+    private static int dividerPosition;
+    private static int charAfterDividerPosition;
+
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_DONE = "done";
+    private static final String COMMAND_BYE = "bye";
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+    private static final String COMMAND_DELETE = "delete";
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+    private static final ArrayList<Task> taskArrayList = new ArrayList<>();
+    public static final String FILEPATH = "data/duke.txt";
+    public static final String FOLDERPATH = "data";
+
+    private static void printLogo() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+    }
+
+    private static void printGreeting(String s, String s2) {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println(s);
+        System.out.println(s2);
+    }
+
+    private static String getUserInput() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Enter command: ");
+        String input = SCANNER.nextLine();
+        // removes whitespace from user input
```

Comment:
This comment seems a bit redundant. Usually, comments are used to explain very complex logic. In this case, it is quite clear what trim() does.